### DISCUSSION TITLE
Date matching from content (Edit: already implemented)

### DISCUSSION THREAD

**Discussion by shamoon** [jonaswinkler/paperless-ng#188] [Feature Requests]
Hi again, so this would be especially useful with #187 but even without would be great, the idea would be to use OCR content and pull out any date. If this is implemented before/without suggestions #187 it could rank dates and choose the top (maybe a date that occurs more than once or the first one). Again this is something I’ve seen elsewhere and was pretty useful.

Appreciate peoples thoughts.

Edit: it's already doing this isnt it! Maybe just some of my docs it didn't pick up. Sorry didn't see in the documenation

**Comment by jonaswinkler** [comment]
Might be that it's not properly documented. But yeah, that's always been there. Sometimes OCR messes up the dates. Sometimes some weird phone numbers get picked up as dates, and sometimes my date of birth gets picked up as the date created instead of the actual date that's also on the document.

But overall, it's working out for me okay-ish.

**Comment by totti4ever** [comment]
What might indeed be an idea to improve this is to show the other, lower-ranked and therefore not chosen, dates for a quick pick of the right date, e.g. in the case the birth date has been chosen. Docspell does this and it makes setting the right date way quicker....but then again this is mainly helpful if you are having heaps of documents to validate and correct, which is the case for most people only at the beginning I assume

This is more or less the same which has been suggested in #187 for tags and correspondants, indeed.

I suggest, closing this discussion and adding the document date to #187

**Comment by shamoon** [comment]
Yep sorry I meant #187 to include dates too, I wrote both of these =)

**Comment by jonaswinkler** [comment]
The date detection is pretty damn stupid right now and just selects the first things that remotely looks like a date. There's no ranking.

I'll consider adding this in the future, sounds useful!

**Comment by tanderson1992** [comment]
Adding my 2 cents. I just came over from paperless, and the behavior seems similar here. Most of my documents are US-generated and use the MM/DD/YY or YYYY format. The frontend can display in this format, but the consumer seems to read "10/06/2020" and interpret it as 10 June 2020 instead of 6 October 2020. This is why I'm apparently one of the few people who actually liked the filename guesswork. After 6 months on paperless even my wife is trained to name documents as YYYYMMDDZ - Correspondent - Title - tags.pdf. But that's another discussion....
